### PR TITLE
WP-1101 added FieldLabel visual test

### PR DESCRIPTION
**Pull request by sergeychernyshev**
Visual test for FieldLabel (including focus passing to another element).

JIRA: https://meetup.atlassian.net/browse/WP-1101

Also added support for puppeteer control before screenshotting.

### CODE REVIEW

**Comment by sergeychernyshev** [comment]
Merged two tests - luckily only one of them requires interaction so doing one click on the whole page was good.

File: packages/swarm-components/src/FieldLabel.test.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+	const testCases = [
+		['Label alone', <FieldLabel key="alone">Field Label</FieldLabel>],
+		[
+			'Pass focus',
+			<div key="focus">
+				<FieldLabel id="swarm-external-label" htmlFor="swarm-checkbox">
+					External checkbox Label
+				</FieldLabel>
+				<Checkbox id="swarm-checkbox">Checkbox&apos;s own text</Checkbox>
```
Comment (sadafie) [inline packages/swarm-components/src/FieldLabel.test.jsx]: Why does `FieldLabel` have a `pass focus` state? From my understanding `FieldLabel` should be completely independent of the form element. Is that not the case?

Comment (sergeychernyshev) [reply packages/swarm-components/src/FieldLabel.test.jsx]: It's HTML core functionality of labels (not React state) where click on the label should focus the element specified in React's `htmlFor` / HTML's `for` attribute.

Comment (sergeychernyshev) [reply packages/swarm-components/src/FieldLabel.test.jsx]: So it is a functional test (not necessarily purely visual) that ensures that focus is passed (see blue border around checkbox in a screenshot).

File: packages/swarm-components/src/testUtils/screenRenderer.js [hunk 2]
```diff
@@ -111,6 +111,10 @@ class screenRenderer {
 		this.log(`Testing: ${testUrl}`);
 		await page.goto(testUrl);
 
+		if (screenshotConfig.beforeScreenshot) {
```
Comment (sadafie) [inline packages/swarm-components/src/testUtils/screenRenderer.js]: this looks unused now

Comment (sergeychernyshev) [reply packages/swarm-components/src/testUtils/screenRenderer.js]: Yep, but can be used in the future as needed ;)

Comment (sadafie) [reply packages/swarm-components/src/testUtils/screenRenderer.js]: hm, how can we document it so it's not a hidden feature? Not sure anyone would realize this is here.

Comment (sergeychernyshev) [reply packages/swarm-components/src/testUtils/screenRenderer.js]: Added the docs

**Comment by sadafie** [comment]
@sergeychernyshev if you pull the latest master, you can convert these to jest snapshots. it's very simple, you can use Button.test.jsx as an example.

**Comment by sergeychernyshev** [comment]
Replaced with Jest snapshot: https://github.com/meetup/swarm-ui/pull/85

**Pull request closed without merging**